src/linear_algebra/bilinear_form.lean

```diff
@@ -1680,10 +1681,27 @@ theorem _root_.matrix.nondegenerate.to_bilin' {M : matrix ι ι R₃} (h : M.non
 λ x hx, h.eq_zero_of_ortho (λ y,
   by simpa only [to_bilin'_apply, dot_product, mul_vec, finset.mul_sum, mul_assoc] using hx y)
 
+lemma _root_.matrix.nondegenerate_iff_to_bilin' {M : matrix ι ι R₃} :
+  M.nondegenerate ↔ (matrix.to_bilin' M).nondegenerate :=
+begin
+  refine ⟨λ h, matrix.nondegenerate.to_bilin' h, λ h, λ v hv, _⟩,
+  replace h := h v,
+  simp_rw [matrix.to_bilin'_apply'] at h,
+  exact h hv
+end
+
 theorem nondegenerate_of_det_ne_zero' (M : matrix ι ι A) (h : M.det ≠ 0) :
   (to_bilin' M).nondegenerate :=
 (matrix.nondegenerate_of_det_ne_zero h).to_bilin'
 
+lemma to_bilin.nondegenerate_iff_det_ne_zero' {M : matrix ι ι A} :
```

Comment:
Dot notation doesn't make a whole lot of sense here:
```suggestion
lemma to_bilin'_nondegenerate_iff_det_ne_zero {M : matrix ι ι A} :
```
feels like a better name to me.